pkg/controller.v1beta1/suggestion/suggestionclient/suggestionclient_test.go

```diff
@@ -539,6 +539,138 @@ func TestConvertParameterType(t *testing.T) {
 	}
 }
 
+func TestConvertFeasibleSpace(t *testing.T) {
+
+	tcs := []struct {
+		inFeasibleSpace       experimentsv1beta1.FeasibleSpace
+		expectedFeasibleSpace *suggestionapi.FeasibleSpace
+		testDescription       string
+	}{
+		{
+			inFeasibleSpace: experimentsv1beta1.FeasibleSpace{
+				Max:          "10",
+				Min:          "1",
+				List:         []string{"1", "2", "3"},
+				Step:         "1",
+				Distribution: experimentsv1beta1.DistributionUniform,
+			},
+			expectedFeasibleSpace: &suggestionapi.FeasibleSpace{
+				Max:          "10",
+				Min:          "1",
+				List:         []string{"1", "2", "3"},
+				Step:         "1",
+				Distribution: suggestionapi.Distribution_UNIFORM,
+			},
+			testDescription: "Convert feasible space with uniform distribution",
+		},
+		{
+			inFeasibleSpace: experimentsv1beta1.FeasibleSpace{
+				Max:          "100",
+				Min:          "10",
+				List:         nil,
+				Step:         "10",
+				Distribution: experimentsv1beta1.DistributionLogUniform,
+			},
+			expectedFeasibleSpace: &suggestionapi.FeasibleSpace{
+				Max:          "100",
+				Min:          "10",
+				List:         nil,
+				Step:         "10",
+				Distribution: suggestionapi.Distribution_LOG_UNIFORM,
+			},
+			testDescription: "Convert feasible space with log-uniform distribution",
+		},
+		{
+			inFeasibleSpace: experimentsv1beta1.FeasibleSpace{
+				Max:          "1.0",
+				Min:          "0.0",
+				List:         nil,
+				Step:         "0.1",
+				Distribution: experimentsv1beta1.DistributionNormal,
+			},
+			expectedFeasibleSpace: &suggestionapi.FeasibleSpace{
+				Max:          "1.0",
+				Min:          "0.0",
+				List:         nil,
+				Step:         "0.1",
+				Distribution: suggestionapi.Distribution_NORMAL,
+			},
+			testDescription: "Convert feasible space with normal distribution",
+		},
+	}
+
+	for _, tc := range tcs {
+		actualFeasibleSpace := convertFeasibleSpace(tc.inFeasibleSpace)
+		if !compareFeasibleSpaces(actualFeasibleSpace, tc.expectedFeasibleSpace) {
```

Comment:
Instead of this custom function, could you use the `https://github.com/google/go-cmp`?
You can find some guidance for developing here: https://github.com/kubeflow/katib/blob/master/docs/developer-guide.md#go-development

Reply:
Thanks, removed the helper function and used `cmp.Diff` instead.